pennylane/devices/qubit/apply_operation.py

```diff
@@ -261,21 +261,14 @@ def apply_mid_measure(
     if is_state_batched:
         raise ValueError("MidMeasureMP cannot be applied to batched states.")
     if not np.allclose(np.linalg.norm(state), 1.0):
-        mid_measurements[op] = 0
+        mid_measurements[op] = -1
         return np.zeros_like(state)
     wire = op.wires
     probs = qml.devices.qubit.measure(qml.probs(wire), state)
-
-    try:  # pragma: no cover
-        sample = np.random.binomial(1, probs[1])
-    except ValueError as e:  # pragma: no cover
-        if probs[1] > 1:  # MachEps error, safe to catch
-            sample = np.random.binomial(1, np.round(probs[1], 15))
-        else:  # Other general error, continue to fail
-            raise e
-
+    sample = np.random.binomial(1, probs[1])
```

Comment:
Have we fixed the mach eps error with this change somehow?

Reply:
I was not aware of this change. I'm not sure how it got deleted TBH, maybe a `git merge` + conflict resolution mistake.